{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "d6886d3d",
   "metadata": {},
   "source": [
    "# Biological DataBase Design with SQLite ðŸ§¬ðŸ—‚\n",
    "\n",
    "### Bio Eng C131 | Lab 4 Adaptation | Fall 2020\n",
    "### Cameron Hirsh  "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1524f796",
   "metadata": {},
   "source": [
    "## Overview\n",
    "\n",
    "#### Background\n",
    "\n",
    "The lab I'm working in studies metabolism. Over the years, many students and post-docs have studied enzymes from organisms across the tree of life, from flies, worms, mice, and E. coli. My PI wants me to build a database of these enzymes that future researchers in the lab can use going forward. \n",
    "\n",
    "We will focus on three pathways: \n",
    "\n",
    "1. glycolysis \n",
    "2. the citric acid cycle \n",
    "3. pentose phosphate pathway.\n",
    "\n",
    "We will consider genes in these pathways from:\n",
    "\n",
    "1. Drosophila\n",
    "2. E. coli\n",
    "3. humans\n",
    "\n",
    "Researchers want to be able to query for:\n",
    "* a gene, and retreive information on all enzymes associated with the gene (in regards to the 3 metabolic pathways)\n",
    "* a pathway, and retreive information on all genes associated with the pathway.\n",
    "\n",
    "#### Approach\n",
    "\n",
    "To do this, I will construct a database using remote data pulled from [KEGG](https://www.genome.jp/kegg-bin/show_pathway?map00010), [NCBI](https://www.ncbi.nlm.nih.gov/) and then join the information into a structure that supports these types of queries in a simple and readable fashion. I will construct this database using SQLite supported by python.\n",
    "\n",
    "I will have a:\n",
    "\n",
    "1. _pathway table:_ This table will detail the three metabolic pathways we are studying including the name of the pathway and what the pathway entails.\n",
    "\n",
    "\n",
    "| id | Pathway Name | Decription |\n",
    "| --- | --- | --- |\n",
    "| primary key | name | description |\n",
    "\n",
    "2. _enzyme table:_ data from 4 enzymes from each pathway.\n",
    "\n",
    "| id | Enzyme Name | Function | EC # | pathway |\n",
    "| --- | --- | --- | --- | --- |\n",
    "| id | name | function | number | pathway |\n",
    "\n",
    "3. _gene table:_ genes representing each of the 12 enzymes for each of 3 organisms.\n",
    "\n",
    "| id | Gene Name | Organism | Enzyme | Enzyme ID | Sequence |\n",
    "| --- | --- | --- | --- | --- | --- |\n",
    "| id | name | organism | enzyme | id for enzyme | Nucleic Acids |\n",
    "\n",
    "I will then Join these tables to allow for querrying the desired information simply and clearly."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3eadb33e",
   "metadata": {},
   "source": [
    "## Setup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "ea3e5ca6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Setup\n",
    "\n",
    "import sqlite3\n",
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "\n",
    "conn = sqlite3.connect('metabolism.db')\n",
    "c = conn.cursor()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2927c276",
   "metadata": {},
   "source": [
    "## Step 1: Pathway Information\n",
    "\n",
    "Using KEGG adn NCBI to create custom pathway data file, similar to data file that may be found from a remote source, I will parse the data into the pathway data table."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a590a1ef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [id, name, description]\n",
       "Index: []"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Pathway table\n",
    "\n",
    "# Clear previous Table data\n",
    "c.execute(\"DELETE FROM pathway WHERE id>0;\")\n",
    "\n",
    "c.execute('''CREATE TABLE IF NOT EXISTS pathway\n",
    "                 (id INT PRIMARY KEY ASC, name TEXT, description TEXT )''')\n",
    "\n",
    "pd.read_sql_query(\"SELECT * FROM pathway;\", conn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "763d02cc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style  type=\"text/css\" >\n",
       "#T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7row0_col0,#T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7row0_col1,#T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7row0_col2,#T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7row1_col0,#T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7row1_col1,#T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7row1_col2,#T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7row2_col0,#T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7row2_col1,#T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7row2_col2{\n",
       "            text-align:  left;\n",
       "        }</style><table id=\"T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7\" ><thead>    <tr>        <th class=\"blank level0\" ></th>        <th class=\"col_heading level0 col0\" >id</th>        <th class=\"col_heading level0 col1\" >name</th>        <th class=\"col_heading level0 col2\" >description</th>    </tr></thead><tbody>\n",
       "                <tr>\n",
       "                        <th id=\"T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "                        <td id=\"T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7row0_col0\" class=\"data row0 col0\" >1</td>\n",
       "                        <td id=\"T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7row0_col1\" class=\"data row0 col1\" >Glycolysis</td>\n",
       "                        <td id=\"T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7row0_col2\" class=\"data row0 col2\" >Linear metabolic pathway of enzyme-catalyzed reactions that convert glucose into two molecules of pyruvate in the presence of oxygen or into two molecules of lactate in the absence of oxygen.</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                        <th id=\"T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "                        <td id=\"T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7row1_col0\" class=\"data row1 col0\" >2</td>\n",
       "                        <td id=\"T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7row1_col1\" class=\"data row1 col1\" >Citric Acid Cycle (TCA, Krebs)</td>\n",
       "                        <td id=\"T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7row1_col2\" class=\"data row1 col2\" >Series of chemical reactions to release stored energy through the oxidation of acetyl-CoA derived from carbohydrates, fats, and proteins</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                        <th id=\"T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "                        <td id=\"T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7row2_col0\" class=\"data row2 col0\" >3</td>\n",
       "                        <td id=\"T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7row2_col1\" class=\"data row2 col1\" >Pentose Phosphate</td>\n",
       "                        <td id=\"T_4f6c3c88_1eeb_11ec_b29f_c4b301c47be7row2_col2\" class=\"data row2 col2\" >Parallel to glycolysis. It generates NADPH and pentoses as well as ribose 5-phosphate, a precursor for the synthesis of nucleotides. While the pentose phosphate pathway does involve oxidation of glucose, its primary role is anabolic rather than catabolic. </td>\n",
       "            </tr>\n",
       "    </tbody></table>"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x1266753c8>"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Populate Pathway Table\n",
    "\n",
    "# Add data using SQL Query\n",
    "pathway_file = open(\"pathway_data.txt\")\n",
    "\n",
    "counter = 1\n",
    "for i, line in enumerate(pathway_file):\n",
    "    \n",
    "    line = line.replace('\\n', '')\n",
    "    \n",
    "    \n",
    "    if i < 5:\n",
    "        continue\n",
    "    elif i % 3 == 2:\n",
    "        name = line\n",
    "    elif i % 3 == 0:\n",
    "        description = line\n",
    "    else:\n",
    "        c.execute(\"INSERT INTO pathway (id, name, description) VALUES (\" + str(counter) + \", '\" + name + \"', '\" + description + \"' )\")\n",
    "        counter += 1\n",
    "        \n",
    "\n",
    "# View the Table\n",
    "pathway_df = pd.read_sql_query(\"SELECT * FROM pathway;\", conn)\n",
    "pathway_df = pathway_df.style.set_properties(**{'text-align': 'left'})\n",
    "pathway_df\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "93fdf89b",
   "metadata": {},
   "source": [
    "looks good, lets grab the enzyme data. "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0a8fd73a",
   "metadata": {},
   "source": [
    "## Step 2: Enzyme Data From Pathways\n",
    "\n",
    "Using KEGG and NCBI to create custom enzyme data file, similar to data file that may be found from a remote source, I will parse the data into the enzyme data table."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "95f98a13",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>name</th>\n",
       "      <th>pathway</th>\n",
       "      <th>function</th>\n",
       "      <th>ecnum</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [id, name, pathway, function, ecnum]\n",
       "Index: []"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Enzyme table\n",
    "\n",
    "# Clear previous Table data\n",
    "try :\n",
    "    c.execute(\"DELETE FROM enzyme WHERE id>0;\")\n",
    "except:\n",
    "    pass\n",
    "\n",
    "\n",
    "c.execute('''CREATE TABLE IF NOT EXISTS enzyme\n",
    "                (id INT PRIMARY KEY ASC, name TEXT, pathway TEXT, function TEXT, ecnum TEXT )''')\n",
    "\n",
    "pd.read_sql_query(\"SELECT * FROM enzyme;\", conn)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e6b2bead",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>name</th>\n",
       "      <th>pathway</th>\n",
       "      <th>function</th>\n",
       "      <th>ecnum</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>hexokinase</td>\n",
       "      <td>glycolysis</td>\n",
       "      <td>phosphorylates hexoses (six-carbon sugars), fo...</td>\n",
       "      <td>2.7.1.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>phosphoglucose isomerase</td>\n",
       "      <td>glycolysis</td>\n",
       "      <td>interconverts glucose-6-phosphate (G6P) and fr...</td>\n",
       "      <td>5.3.1.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>phosphofructokinase</td>\n",
       "      <td>glycolysis</td>\n",
       "      <td>phosphorylates fructose 6-phosphate.</td>\n",
       "      <td>2.7.1.11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>aldolase</td>\n",
       "      <td>glycolysis</td>\n",
       "      <td>catalyzes a reversible reaction that splits th...</td>\n",
       "      <td>4.1.2.13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>Citrate synthase</td>\n",
       "      <td>citric acid cycle</td>\n",
       "      <td>catalyze the conversion of acetyl-CoA [H3CC(=O...</td>\n",
       "      <td>2.3.3.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>Aconitase</td>\n",
       "      <td>citric acid cycle</td>\n",
       "      <td>enzyme that catalyses the stereo-specific isom...</td>\n",
       "      <td>4.2.1.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>7</td>\n",
       "      <td>Isocitrate dehydrogenase</td>\n",
       "      <td>citric acid cycle</td>\n",
       "      <td>catalyzes the oxidative decarboxylation of iso...</td>\n",
       "      <td>1.1.1.42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>8</td>\n",
       "      <td>Fumarase</td>\n",
       "      <td>citric acid cycle</td>\n",
       "      <td>catalyzes the reversible hydration/dehydration...</td>\n",
       "      <td>4.2.1.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>G6PD</td>\n",
       "      <td>pentose phosphate</td>\n",
       "      <td>catalyzes the chemical reaction: D-glucose 6-p...</td>\n",
       "      <td>2.3.3.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>6PGD</td>\n",
       "      <td>pentose phosphate</td>\n",
       "      <td>forms ribulose 5-phosphate from 6-phosphogluco...</td>\n",
       "      <td>1.1.1.44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>11</td>\n",
       "      <td>Gluconolactonase</td>\n",
       "      <td>pentose phosphate</td>\n",
       "      <td>This enzyme belongs to the family of hydrolase...</td>\n",
       "      <td>3.1.1.17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>12</td>\n",
       "      <td>Transaldolase</td>\n",
       "      <td>pentose phosphate</td>\n",
       "      <td>transaldolase:catalyzes the following reaction...</td>\n",
       "      <td>2.2.1.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    id                      name            pathway  \\\n",
       "0    1                hexokinase         glycolysis   \n",
       "1    2  phosphoglucose isomerase         glycolysis   \n",
       "2    3       phosphofructokinase         glycolysis   \n",
       "3    4                  aldolase         glycolysis   \n",
       "4    5          Citrate synthase  citric acid cycle   \n",
       "5    6                 Aconitase  citric acid cycle   \n",
       "6    7  Isocitrate dehydrogenase  citric acid cycle   \n",
       "7    8                  Fumarase  citric acid cycle   \n",
       "8    9                      G6PD  pentose phosphate   \n",
       "9   10                      6PGD  pentose phosphate   \n",
       "10  11          Gluconolactonase  pentose phosphate   \n",
       "11  12             Transaldolase  pentose phosphate   \n",
       "\n",
       "                                             function     ecnum  \n",
       "0   phosphorylates hexoses (six-carbon sugars), fo...   2.7.1.1  \n",
       "1   interconverts glucose-6-phosphate (G6P) and fr...   5.3.1.9  \n",
       "2                phosphorylates fructose 6-phosphate.  2.7.1.11  \n",
       "3   catalyzes a reversible reaction that splits th...  4.1.2.13  \n",
       "4   catalyze the conversion of acetyl-CoA [H3CC(=O...   2.3.3.1  \n",
       "5   enzyme that catalyses the stereo-specific isom...   4.2.1.3  \n",
       "6   catalyzes the oxidative decarboxylation of iso...  1.1.1.42  \n",
       "7   catalyzes the reversible hydration/dehydration...   4.2.1.2  \n",
       "8   catalyzes the chemical reaction: D-glucose 6-p...   2.3.3.1  \n",
       "9   forms ribulose 5-phosphate from 6-phosphogluco...  1.1.1.44  \n",
       "10  This enzyme belongs to the family of hydrolase...  3.1.1.17  \n",
       "11  transaldolase:catalyzes the following reaction...   2.2.1.2  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Populate Enzyme Table\n",
    "\n",
    "# Add data using SQL Query\n",
    "\n",
    "enzyme_file = open(\"enzyme_data.txt\")\n",
    "\n",
    "counter = 1\n",
    "for i, line in enumerate(enzyme_file):\n",
    "    \n",
    "    line = line.replace('\\n', '')\n",
    "    \n",
    "    \n",
    "    if i < 7:\n",
    "        continue\n",
    "    elif i % 5 == 2:\n",
    "        name = line\n",
    "    elif i % 5 == 3:\n",
    "        pathway = line\n",
    "    elif i % 5 == 4:\n",
    "        function = line\n",
    "    elif i % 5 == 0:\n",
    "        ecnum = line\n",
    "    else:\n",
    "        c.execute(\"INSERT INTO enzyme (id, name, pathway, function, ecnum) VALUES (\" \n",
    "                  + str(counter) + \", '\" + name + \"', '\" + pathway + \"', '\" + function + \"', '\" + ecnum + \"' )\")\n",
    "        counter += 1\n",
    "        \n",
    "\n",
    "\n",
    "# View the Table\n",
    "pd.read_sql_query(\"SELECT * FROM enzyme;\", conn)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "43edffe7",
   "metadata": {},
   "source": [
    "## Step 3: Gather & Consolidate Remote Gene Data"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "37d76413",
   "metadata": {},
   "source": [
    "Pull remotely 3 Fasta Format Files of nucleic acid sequences representing most relevant search results from [NCBi database](usinghttps://www.ncbi.nlm.nih.gov/) to find genes for the selected enzymes for each of the 3 organisms. Will parse each fasta file into database."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "5228193d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>name</th>\n",
       "      <th>organism</th>\n",
       "      <th>enzyme</th>\n",
       "      <th>enzyme_id</th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [id, name, organism, enzyme, enzyme_id, sequence]\n",
       "Index: []"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Gene table\n",
    "\n",
    "# Clear previous Table data\n",
    "try:\n",
    "    c.execute(\"DELETE FROM gene WHERE id>0;\")\n",
    "except:\n",
    "    pass\n",
    "\n",
    "c.execute('''CREATE TABLE IF NOT EXISTS gene\n",
    "                (id INT PRIMARY KEY ASC, name TEXT, organism TEXT, enzyme TEXT, enzyme_id INT, sequence TEXT )''')\n",
    "\n",
    "pd.read_sql_query(\"SELECT * FROM gene;\", conn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "3268f387",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Enzyme Matching Functionality\n",
    "\n",
    "enzyme_df = pd.read_sql_query(\"SELECT * FROM enzyme;\", conn)\n",
    "enzymes = enzyme_df['name']\n",
    "\n",
    "# takes in messy enzyme name and returns enzyme id for a match\n",
    "def helper(enzyme_name):\n",
    "    for i in range(len(enzymes)):\n",
    "        if enzymes[i].lower() in enzyme_name.lower():\n",
    "            return i + 1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "a8b89435",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>name</th>\n",
       "      <th>organism</th>\n",
       "      <th>enzyme</th>\n",
       "      <th>enzyme_id</th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>M75126.1</td>\n",
       "      <td>Homo Sapien</td>\n",
       "      <td>hexokinase 1 (HK1) mRNA, complete cds</td>\n",
       "      <td>1</td>\n",
       "      <td>CCGCCGGAGGACCACGGCTCGCCAGGGCTGCGGAGGACCGACCGTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>XX_0</td>\n",
       "      <td>Homo Sapien</td>\n",
       "      <td>phosphoglucose isomerase (PGI)</td>\n",
       "      <td>2</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>M64784.1</td>\n",
       "      <td>Homo Sapien</td>\n",
       "      <td>phosphofructokinase (PFK) mRNA, complete cds</td>\n",
       "      <td>3</td>\n",
       "      <td>GAATTCCGAAGTACTTGGAAGAGATCGCCACACAGATGCGCACCAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>XX_0</td>\n",
       "      <td>Homo Sapien</td>\n",
       "      <td>aldolase</td>\n",
       "      <td>4</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>AF053631.1</td>\n",
       "      <td>Homo Sapien</td>\n",
       "      <td>citrate synthase mRNA, complete cds</td>\n",
       "      <td>5</td>\n",
       "      <td>CCGGGCTGGGCGCCGCCGCCGGTTCGTCTACTCTTTCCTTCAGCCG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>BC014092.2</td>\n",
       "      <td>Homo Sapien</td>\n",
       "      <td>aconitase 2, mitochondrial, mRNA (cDNA clone M...</td>\n",
       "      <td>6</td>\n",
       "      <td>CTCATCTTTGTCAGTGCACAAAATGGCGCCCTACAGCCTACTGGTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>7</td>\n",
       "      <td>KU639670.1</td>\n",
       "      <td>Homo Sapien</td>\n",
       "      <td>voucher NGX277 isocitrate dehydrogenase (IDH2)...</td>\n",
       "      <td>7</td>\n",
       "      <td>TCCCAATGGAACTATCCGGAACATCCTGGGGGGGACTGTCTTCCGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>8</td>\n",
       "      <td>XX_0</td>\n",
       "      <td>Homo Sapien</td>\n",
       "      <td>Fumarase</td>\n",
       "      <td>8</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>MT640143.1</td>\n",
       "      <td>Homo Sapien</td>\n",
       "      <td>isolate ml2608 Yunnan G6PD (G6PD) gene, partia...</td>\n",
       "      <td>9</td>\n",
       "      <td>ATGGCAGAGCAGGTGGCCCTGAGCCGGACCCAGGTGTGCGGGATCC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>NM_002631.4</td>\n",
       "      <td>Homo Sapien</td>\n",
       "      <td>phosphogluconate dehydrogenase (PGD), transcri...</td>\n",
       "      <td>10</td>\n",
       "      <td>CCCTCACTCGTCCTCCGCGCGTCGCCGCTCTTCGGTTCTGCTCTGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>11</td>\n",
       "      <td>XX_0</td>\n",
       "      <td>Homo Sapien</td>\n",
       "      <td>Gluconolactonase</td>\n",
       "      <td>11</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>12</td>\n",
       "      <td>NM_006755.2</td>\n",
       "      <td>Homo Sapien</td>\n",
       "      <td>transaldolase 1 (TALDO1), mRNA</td>\n",
       "      <td>12</td>\n",
       "      <td>AGACCCCTCGGTCTTGCTATGTCGAGCTCACCCGTGAAGCGTCAGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>13</td>\n",
       "      <td>NM_080109.4</td>\n",
       "      <td>Drosophila</td>\n",
       "      <td>melanogaster hexokinase A, transcript variant...</td>\n",
       "      <td>1</td>\n",
       "      <td>TTAATTTCGTAAAAAACGTTCCGAGCGTTTGGACGCGTGCAGCGGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>14</td>\n",
       "      <td>MG684428.1</td>\n",
       "      <td>Drosophila</td>\n",
       "      <td>sulfurigaster phosphoglucose isomerase (Pgi) ...</td>\n",
       "      <td>2</td>\n",
       "      <td>AGGATGATGCAGTGGCCAAGCACTTTGTGGCTCTATCCACTAACAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>15</td>\n",
       "      <td>NM_078952.3</td>\n",
       "      <td>Drosophila</td>\n",
       "      <td>melanogaster phosphofructokinase, transcript ...</td>\n",
       "      <td>3</td>\n",
       "      <td>CTAAAATGTCAAAAATCATAAAAAAAAAATGAGTTCAGAAAATGAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>16</td>\n",
       "      <td>NM_001300612.1</td>\n",
       "      <td>Drosophila</td>\n",
       "      <td>melanogaster aldolase 1, transcript variant M...</td>\n",
       "      <td>4</td>\n",
       "      <td>AGTCGGTTGCCCAACACCGTCGGCGACGACCTCCAAACAACCTTTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>17</td>\n",
       "      <td>XX_0</td>\n",
       "      <td>Drosophila</td>\n",
       "      <td>Citrate synthase</td>\n",
       "      <td>5</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>18</td>\n",
       "      <td>NM_001299190.1</td>\n",
       "      <td>Drosophila</td>\n",
       "      <td>melanogaster mitochondrial aconitase 1, trans...</td>\n",
       "      <td>6</td>\n",
       "      <td>AATTTTGGCGGGAAACAAGCCGGCGGAACATTGCTGACGCGTATGC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>19</td>\n",
       "      <td>NM_001144438.2</td>\n",
       "      <td>Drosophila</td>\n",
       "      <td>melanogaster isocitrate dehydrogenase, transc...</td>\n",
       "      <td>7</td>\n",
       "      <td>ACGCCGATTGCGCAACAATTTGACGATTGTACGCCACCCAAGAGAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>20</td>\n",
       "      <td>NM_167081.3</td>\n",
       "      <td>Drosophila</td>\n",
       "      <td>melanogaster fumarase 1, transcript variant B...</td>\n",
       "      <td>8</td>\n",
       "      <td>CTCAGTTAACATCGAGCAGCAGCCGTCGTCGAGTGGTCGTCCTCTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>21</td>\n",
       "      <td>AY364534.1</td>\n",
       "      <td>Drosophila</td>\n",
       "      <td>mojavensis strain MJS 71 glucose-6-phosphate ...</td>\n",
       "      <td>9</td>\n",
       "      <td>GGCTGGAATCGCGTGATCGTCGAGAAGCCCTTCGGGCGAGATGATG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>22</td>\n",
       "      <td>XX_0</td>\n",
       "      <td>Drosophila</td>\n",
       "      <td>6PGD</td>\n",
       "      <td>10</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>23</td>\n",
       "      <td>XX_0</td>\n",
       "      <td>Drosophila</td>\n",
       "      <td>Gluconolactonase</td>\n",
       "      <td>11</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>24</td>\n",
       "      <td>NM_079111.3</td>\n",
       "      <td>Drosophila</td>\n",
       "      <td>melanogaster transaldolase (Taldo), mRNA</td>\n",
       "      <td>12</td>\n",
       "      <td>TATTGCATTGAAGTTGGGCAATGGGAAGTGATCGCACGTTAAAAAA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>25</td>\n",
       "      <td>XX_0</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>Hexokinase</td>\n",
       "      <td>1</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>26</td>\n",
       "      <td>AJ629834.1</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>partial pgi gene for phosphoglucose isomerase,...</td>\n",
       "      <td>2</td>\n",
       "      <td>GGTATTGATACTGCCAACATGTTCGAGTTCTGGGACTGGGTCGGCG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>27</td>\n",
       "      <td>XX_0</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>phosphofructokinase (PFK)</td>\n",
       "      <td>3</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>28</td>\n",
       "      <td>XX_0</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>aldolase</td>\n",
       "      <td>4</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>29</td>\n",
       "      <td>EU904044.1</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>strain TW14359 citrate synthase (ECs0745) gene...</td>\n",
       "      <td>5</td>\n",
       "      <td>ATGGCTGATACAAAAGCAAAACTCACCCTCAACGGGGACACAGCTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>30</td>\n",
       "      <td>XX_0</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>aconitase</td>\n",
       "      <td>6</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>31</td>\n",
       "      <td>AJ629672.1</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>partial icd gene for isocitrate dehydrogenase,...</td>\n",
       "      <td>7</td>\n",
       "      <td>GTTCCTGAAAATCCGATTATCCCTTACATTGAAGGTGATGGAATCG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>32</td>\n",
       "      <td>XX_0</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>fumarase</td>\n",
       "      <td>8</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>33</td>\n",
       "      <td>EU899544.1</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>strain TW14359 glucose-6-phosphate dehydrogena...</td>\n",
       "      <td>9</td>\n",
       "      <td>ATGGCGGTAACGCAAACAGCCCAGGTCTGTGACTTGGTCATTTTCG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>34</td>\n",
       "      <td>M23181.1</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>6-phosphogluconate dehydrogenase gene, complet...</td>\n",
       "      <td>10</td>\n",
       "      <td>AGATCTCACTAAAAACTGGGGATAACGCCTTAAATGGCGAAGAAAC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>35</td>\n",
       "      <td>XX_0</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>Gluconolactonase</td>\n",
       "      <td>11</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>36</td>\n",
       "      <td>D13161.1</td>\n",
       "      <td>E. coli</td>\n",
       "      <td>talB gene for transaldolase, complete cds</td>\n",
       "      <td>12</td>\n",
       "      <td>ATGACGGACAAATTGACCTCCCTTCGTCAGTACACCACCGTAGTGG...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    id            name     organism  \\\n",
       "0    1        M75126.1  Homo Sapien   \n",
       "1    2            XX_0  Homo Sapien   \n",
       "2    3        M64784.1  Homo Sapien   \n",
       "3    4            XX_0  Homo Sapien   \n",
       "4    5      AF053631.1  Homo Sapien   \n",
       "5    6      BC014092.2  Homo Sapien   \n",
       "6    7      KU639670.1  Homo Sapien   \n",
       "7    8            XX_0  Homo Sapien   \n",
       "8    9      MT640143.1  Homo Sapien   \n",
       "9   10     NM_002631.4  Homo Sapien   \n",
       "10  11            XX_0  Homo Sapien   \n",
       "11  12     NM_006755.2  Homo Sapien   \n",
       "12  13     NM_080109.4   Drosophila   \n",
       "13  14      MG684428.1   Drosophila   \n",
       "14  15     NM_078952.3   Drosophila   \n",
       "15  16  NM_001300612.1   Drosophila   \n",
       "16  17            XX_0   Drosophila   \n",
       "17  18  NM_001299190.1   Drosophila   \n",
       "18  19  NM_001144438.2   Drosophila   \n",
       "19  20     NM_167081.3   Drosophila   \n",
       "20  21      AY364534.1   Drosophila   \n",
       "21  22            XX_0   Drosophila   \n",
       "22  23            XX_0   Drosophila   \n",
       "23  24     NM_079111.3   Drosophila   \n",
       "24  25            XX_0      E. coli   \n",
       "25  26      AJ629834.1      E. coli   \n",
       "26  27            XX_0      E. coli   \n",
       "27  28            XX_0      E. coli   \n",
       "28  29      EU904044.1      E. coli   \n",
       "29  30            XX_0      E. coli   \n",
       "30  31      AJ629672.1      E. coli   \n",
       "31  32            XX_0      E. coli   \n",
       "32  33      EU899544.1      E. coli   \n",
       "33  34        M23181.1      E. coli   \n",
       "34  35            XX_0      E. coli   \n",
       "35  36        D13161.1      E. coli   \n",
       "\n",
       "                                               enzyme  enzyme_id  \\\n",
       "0               hexokinase 1 (HK1) mRNA, complete cds          1   \n",
       "1                      phosphoglucose isomerase (PGI)          2   \n",
       "2        phosphofructokinase (PFK) mRNA, complete cds          3   \n",
       "3                                            aldolase          4   \n",
       "4                 citrate synthase mRNA, complete cds          5   \n",
       "5   aconitase 2, mitochondrial, mRNA (cDNA clone M...          6   \n",
       "6   voucher NGX277 isocitrate dehydrogenase (IDH2)...          7   \n",
       "7                                            Fumarase          8   \n",
       "8   isolate ml2608 Yunnan G6PD (G6PD) gene, partia...          9   \n",
       "9   phosphogluconate dehydrogenase (PGD), transcri...         10   \n",
       "10                                   Gluconolactonase         11   \n",
       "11                     transaldolase 1 (TALDO1), mRNA         12   \n",
       "12   melanogaster hexokinase A, transcript variant...          1   \n",
       "13   sulfurigaster phosphoglucose isomerase (Pgi) ...          2   \n",
       "14   melanogaster phosphofructokinase, transcript ...          3   \n",
       "15   melanogaster aldolase 1, transcript variant M...          4   \n",
       "16                                   Citrate synthase          5   \n",
       "17   melanogaster mitochondrial aconitase 1, trans...          6   \n",
       "18   melanogaster isocitrate dehydrogenase, transc...          7   \n",
       "19   melanogaster fumarase 1, transcript variant B...          8   \n",
       "20   mojavensis strain MJS 71 glucose-6-phosphate ...          9   \n",
       "21                                               6PGD         10   \n",
       "22                                   Gluconolactonase         11   \n",
       "23           melanogaster transaldolase (Taldo), mRNA         12   \n",
       "24                                         Hexokinase          1   \n",
       "25  partial pgi gene for phosphoglucose isomerase,...          2   \n",
       "26                          phosphofructokinase (PFK)          3   \n",
       "27                                           aldolase          4   \n",
       "28  strain TW14359 citrate synthase (ECs0745) gene...          5   \n",
       "29                                          aconitase          6   \n",
       "30  partial icd gene for isocitrate dehydrogenase,...          7   \n",
       "31                                           fumarase          8   \n",
       "32  strain TW14359 glucose-6-phosphate dehydrogena...          9   \n",
       "33  6-phosphogluconate dehydrogenase gene, complet...         10   \n",
       "34                                   Gluconolactonase         11   \n",
       "35          talB gene for transaldolase, complete cds         12   \n",
       "\n",
       "                                             sequence  \n",
       "0   CCGCCGGAGGACCACGGCTCGCCAGGGCTGCGGAGGACCGACCGTC...  \n",
       "1                                                   N  \n",
       "2   GAATTCCGAAGTACTTGGAAGAGATCGCCACACAGATGCGCACCAG...  \n",
       "3                                                   N  \n",
       "4   CCGGGCTGGGCGCCGCCGCCGGTTCGTCTACTCTTTCCTTCAGCCG...  \n",
       "5   CTCATCTTTGTCAGTGCACAAAATGGCGCCCTACAGCCTACTGGTG...  \n",
       "6   TCCCAATGGAACTATCCGGAACATCCTGGGGGGGACTGTCTTCCGG...  \n",
       "7                                                   N  \n",
       "8   ATGGCAGAGCAGGTGGCCCTGAGCCGGACCCAGGTGTGCGGGATCC...  \n",
       "9   CCCTCACTCGTCCTCCGCGCGTCGCCGCTCTTCGGTTCTGCTCTGT...  \n",
       "10                                                  N  \n",
       "11  AGACCCCTCGGTCTTGCTATGTCGAGCTCACCCGTGAAGCGTCAGA...  \n",
       "12  TTAATTTCGTAAAAAACGTTCCGAGCGTTTGGACGCGTGCAGCGGA...  \n",
       "13  AGGATGATGCAGTGGCCAAGCACTTTGTGGCTCTATCCACTAACAA...  \n",
       "14  CTAAAATGTCAAAAATCATAAAAAAAAAATGAGTTCAGAAAATGAA...  \n",
       "15  AGTCGGTTGCCCAACACCGTCGGCGACGACCTCCAAACAACCTTTT...  \n",
       "16                                                  N  \n",
       "17  AATTTTGGCGGGAAACAAGCCGGCGGAACATTGCTGACGCGTATGC...  \n",
       "18  ACGCCGATTGCGCAACAATTTGACGATTGTACGCCACCCAAGAGAG...  \n",
       "19  CTCAGTTAACATCGAGCAGCAGCCGTCGTCGAGTGGTCGTCCTCTG...  \n",
       "20  GGCTGGAATCGCGTGATCGTCGAGAAGCCCTTCGGGCGAGATGATG...  \n",
       "21                                                  N  \n",
       "22                                                  N  \n",
       "23  TATTGCATTGAAGTTGGGCAATGGGAAGTGATCGCACGTTAAAAAA...  \n",
       "24                                                  N  \n",
       "25  GGTATTGATACTGCCAACATGTTCGAGTTCTGGGACTGGGTCGGCG...  \n",
       "26                                                  N  \n",
       "27                                                  N  \n",
       "28  ATGGCTGATACAAAAGCAAAACTCACCCTCAACGGGGACACAGCTG...  \n",
       "29                                                  N  \n",
       "30  GTTCCTGAAAATCCGATTATCCCTTACATTGAAGGTGATGGAATCG...  \n",
       "31                                                  N  \n",
       "32  ATGGCGGTAACGCAAACAGCCCAGGTCTGTGACTTGGTCATTTTCG...  \n",
       "33  AGATCTCACTAAAAACTGGGGATAACGCCTTAAATGGCGAAGAAAC...  \n",
       "34                                                  N  \n",
       "35  ATGACGGACAAATTGACCTCCCTTCGTCAGTACACCACCGTAGTGG...  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Nested Loop to Parse each of 3 files adding FASTA content to Gene Database\n",
    "\n",
    "\n",
    "# Use Seq.IO to Parse Fasta File and add genes to table\n",
    "\n",
    "\n",
    "files = ['homo.fa', 'drosophila.fa', 'ecoli.fa']\n",
    "organisms = ['Homo Sapien', 'Drosophila', 'E. coli']\n",
    "split_by = ['sapiens ', 'Drosophila', 'coli ']\n",
    "\n",
    "gene_id = 1\n",
    "\n",
    "for i in range(0,3):\n",
    "    \n",
    "    for gene in SeqIO.parse(files[i], 'fasta'):\n",
    "        \n",
    "        name = gene.id\n",
    "        organism = organisms[i]\n",
    "        enzyme = gene.description.split(split_by[i])[1]\n",
    "        enzyme_id = helper(enzyme)\n",
    "        seq = str(gene.seq)\n",
    "        \n",
    "        c.execute(\"INSERT INTO gene (id, name, organism, enzyme, enzyme_id, sequence) VALUES (\" \n",
    "                  + str(gene_id) + \", '\" + name + \"', '\" + organism + \"', '\" + enzyme + \"', '\" + str(enzyme_id) + \"', '\" + seq + \"' )\")\n",
    "        gene_id += 1\n",
    "        \n",
    "        \n",
    "# Use SQL to manually update the mis-translated enzymes due to inconsistent naming syntax / missing information\n",
    "c.execute(\"UPDATE gene SET enzyme_id = 10 WHERE id = 10;\")\n",
    "c.execute(\"UPDATE gene SET enzyme_id = 12 WHERE id = 12;\")\n",
    "c.execute(\"UPDATE gene SET enzyme_id = 12 WHERE id = 24;\")\n",
    "c.execute(\"UPDATE gene SET enzyme_id = 9 WHERE id = 33;\")\n",
    "c.execute(\"UPDATE gene SET enzyme_id = 10 WHERE id = 34;\")\n",
    "c.execute(\"UPDATE gene SET enzyme_id = 12 WHERE id = 36;\")   \n",
    "        \n",
    "        \n",
    "pd.read_sql_query(\"SELECT * FROM gene;\", conn)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c99c24ea",
   "metadata": {},
   "source": [
    "Now that we have the database in order we can use SQL JOIN functionality to create tables that can be queried for our intended querying purposes."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eb36fb2e",
   "metadata": {},
   "source": [
    "## Last Step: Join Tables for Desired Query Capability"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d0e8a37f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
